### PR TITLE
PLAT-6350: Lock down storage bucket

### PR DESCRIPTION
**Pull request by fraenkel**
Make bucket settings uniform
Enforce public access, i.e., disabled
Encrypt the bucket with our key

### CODE REVIEW

**Comment by fraenkel** [comment]
Verified a deploy works.
Veriried a backup works.

Upgrading from a previous version is broken. (under discussion)

**Review by ddl-dclegg — APPROVED** [review]

**Comment by ddl-dclegg** [comment]
@fraenkel Will we need to update the [dataclasses, etc](https://github.com/cerebrotech/platform-apps/blob/8b2376cbde0beef97eec51e637868203b69e42ad/src/services/cloud_providers/gcp/resources.py#L315-L388)., in platform-apps after this?

File: service-accounts.tf [hunk 1]
```diff
@@ -19,7 +19,7 @@ resource "google_project_iam_member" "service_account" {
 }
 
 resource "google_project_iam_member" "platform_roles" {
-  for_each = toset(["roles/storage.objectAdmin", "roles/logging.logWriter", "roles/monitoring.metricWriter"])
```
Comment (steved) [inline service-accounts.tf:22]: Why was this removed? Is GCS bucket writing just a given?

**Review by miguelhar — APPROVED** [review]
